### PR TITLE
Implement DarkInspiration node

### PR DESCRIPTION
**Pull request by EmosewaMC**
Will test in a few minutes.

Implements the dark inspiration behavior node.  Since no skills attached to it are cast at the minute, this is effectively a no-op, but is needed for their implementation.

Will be testing that shinobi rank 3 properly executes its behavior.

### CODE REVIEW

File: dGame/dBehaviors/DarkInspirationBehavior.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#include "Entity.h"
+#include "DestroyableComponent.h"
+#include "EntityManager.h"
+#include "BehaviorContext.h"
+
+void DarkInspirationBehavior::Handle(BehaviorContext* context, RakNet::BitStream* bitStream, const BehaviorBranchContext branch) {
+	auto* target = EntityManager::Instance()->GetEntity(branch.target);
+
+	if (target == nullptr) {
+		Game::logger->Log("DamageReductionBehavior", "Failed to find target (%llu)!", branch.target);
```
Comment (aronwk-aaron) [inline dGame/dBehaviors/DarkInspirationBehavior.cpp:13]: Wrong log file name, also, maybe make debug?

**Review by aronwk-aaron — COMMENTED** [review]
One comment, that's all

File: dGame/dBehaviors/DarkInspirationBehavior.cpp (hunk 1 above)
Comment (EmosewaMC) [reply dGame/dBehaviors/DarkInspirationBehavior.cpp:13]: ah right, just copy and paste error

**Review by aronwk-aaron — APPROVED** [review]

File: dGame/dInventory/ItemSetPassiveAbility.cpp [hunk 2]
```diff
@@ -293,8 +293,8 @@ void ItemSetPassiveAbility::OnEnemySmshed() {
 		break;
 	}
 	case ItemSetPassiveAbilityID::ShinobiRank3: {
-		if (equippedCount < 4) return;
-		destroyableComponent->Imagine(3);
+		if (equippedCount < 4 || !target) return;
+		skillComponent->CalculateBehavior(695, 11399, target->GetObjectID());
```
Comment (Wincent01) [inline dGame/dInventory/ItemSetPassiveAbility.cpp:297]: Unfortunate hardcoding. Is there some system behind this which would explain why this is necessary, are all set skills behaviors?

Comment (EmosewaMC) [reply dGame/dInventory/ItemSetPassiveAbility.cpp:297]: There is, but to keep this PR small and just as a proof of concept that the behavior itself works, I only chose to replace this one specifically with the behavior like this.  The actual implementation will be far cleaner

Comment (EmosewaMC) [reply dGame/dInventory/ItemSetPassiveAbility.cpp:297]: I can go into more detail if you want for how this is supposed to be implemented, whether you want all of these to be fixed at once, leave this alone for now and just implement the node, or implement the node and this one check.  any I am fine with

**Review by Jettford — APPROVED** [review]

**Pull request merged**